### ISSUE TITLE
Failure to emulate PMINUB

### ISSUE THREAD

**Issue by jordan9001** [trailofbits/maat#112]
```
[Info] Run 0x43d205: PMINUB XMM0, xmmword ptr [RAX + 0x50]
terminate called after throwing an instance of 'maat::cpu_exception'
  what():  Can't assign 128-bits value to 512-bits register
```
`maat::ir::CPUContext::_check_assignment_size` sees the register as having size `512` inside of `_cpu_ctx.set(inst.out.reg(), pinst.res);`

the `inst.out` knows the correct size of the register to be set is `128`, but `inst.out.reg()` is `ZMM0`, which is `512` bits.

```
inst.out = {
  _val = ZMM0,
  _size = 128,
  type = maat::ir::Param::Type::REG,
  hb = 127,
  lb = 0
}
```

**Comment by jordan9001** [comment]
Maybe `X86_PMINUB_handler` needs to handle different sizes of operands?

**Comment by Boyan-MILANOV** [comment]
> Maybe X86_PMINUB_handler needs to set the result to use the full parent register

I believe so. Usually Maat's p-code CPU handles sub-register assignments but since `PMINUB` uses a custom callback we should take into account the full `ZMM` parent register.

**Comment by Boyan-MILANOV** [comment]
@jordan9001: this will be closed by #117. I'll try to close more issues next week and eventually package everything into a new Maat release
